.github/workflows/publish_maven.yml

```diff
@@ -0,0 +1,127 @@
+        env:
+          MAVEN_USERNAME: ${{ secrets.maven-username }}
+          MAVEN_PASSWORD: ${{ secrets.maven-password }}
+          GPG_KEY_PASS: ${{ secrets.gpg-key-pass }}
+        with:
+          java-version: '11'
+          distribution: 'temurin'
+          server-id: ossrh
+          server-username: MAVEN_USERNAME
+          server-password: MAVEN_PASSWORD
+          gpg-private-key: ${{ secrets.gpg-private-key }}
+          gpg-passphrase: GPG_KEY_PASS
+
+      - name: Download the slsa attestation
+        uses: slsa-framework/slsa-github-generator/.github/actions/secure-download-folder@main
+        with:
+          name: "${{ inputs.provenance-download-name }}"
+          path: slsa-attestations
+          sha256: "${{ inputs.provenance-download-sha256 }}"
+
+      - name: Download the target dir
+        uses: slsa-framework/slsa-github-generator/.github/actions/secure-download-folder@main
+        with:
+          name: target
+          path: ./
+          sha256: "${{ inputs.target-download-sha256 }}"
+
+      - name: Publish to the Maven Central Repository
+        shell: bash
+        env:
+          MAVEN_USERNAME: "${{ secrets.maven-username }}"
+          MAVEN_PASSWORD: "${{ secrets.maven-password }}"
+          GPG_KEY_PASS: "${{ secrets.gpg-key-pass }}"
+          SLSA_DIR: "${{ inputs.provenance-download-name }}"
+          PROVENANCE_FILES: "${{ inputs.provenance-download-name }}"
+        run: |
+          # Build and run custom plugin
+          cd plugin && mvn clean install && cd ..
+          # Re-indexing the secondary jar files for deploy
+          mvn javadoc:jar source:jar
+          echo "find javadoc"
+          find . -name "*javadoc*"
+          echo "end find javadoc"
+          # Retrieve project version
+          VERSION=$(mvn org.apache.maven.plugins:maven-help-plugin:3.2.0:evaluate -Dexpression=project.version -q -DforceStdout)
+          ARTIFACTID=$(mvn org.apache.maven.plugins:maven-help-plugin:3.2.0:evaluate -Dexpression=project.artifactId -q -DforceStdout)
+          # Reset the environment variables add in the base provenance
+          FILES="slsa-attestations/${PROVENANCE_FILES}/${ARTIFACTID}-${VERSION}.jar.intoto.build.slsa"
+          TYPES=slsa
+          CLASSIFIERS=jar.intoto.build
+          # Find all necessary built jar files and attach them to the environment variable deploy
+          # shellcheck disable=SC2044 # We don't spawn a new sub shell since we are appending to global env vars
+          for name in $(find ./ -name "$ARTIFACTID-$VERSION-*.jar")
+          do
+            # shellcheck disable=SC1001 # shellcheck complains over \- but the line does what it should.
+            TARGET=$(echo "${name}" | rev | cut -d\- -f1 | rev)
+            FILES=$FILES,$name
+            TYPES=$TYPES,${TARGET##*.}
+            CLASSIFIERS=$CLASSIFIERS,${TARGET%.*}
+            echo "FILESSS: ${FILES}"
+          done
+          echo "find ./ -name {ARTIFACTID}-{VERSION}-*.jar"
```

Comment:
remove this debugging message? I'm fine keeping it for now if it helps with troubleshooting / logging

Reply:
I would be fine to leave it for now.